Bluetooth: audio: tbs_client: Make GTBS client only build option

MariuszSkamra opened the pull request:
Refactor Kconfig to add a way to enable GTBS only client.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

Thalley commented on subsys/bluetooth/audio/Kconfig.tbs:
```diff
@@ -140,21 +140,23 @@ endif # BT_TBS
 ##################### Call Control Client #####################
 
-config BT_TBS_CLIENT
+config BT_TBS_CLIENT_TBS
```
Hmm, in TBS the server side shall support GTBS, and then 0 or more TBS instances. 

Should that same requirement go for the TBS client? 

In other words, rather than adding this, should `CONFIG_BT_TBS_CLIENT_MAX_TBS_INSTANCES` (if not already) only refer to TBS instances, and not GTBS, so that if `CONFIG_BT_TBS_CLIENT_MAX_TBS_INSTANCES == 0`, then we effectively have a GTBS-only client?

MariuszSkamra replied:
Nice idea, I did not know that `range` can be conditional as well. Could you please take a look at latest revision?

MariuszSkamra replied:
@Thalley I had to fix the `primary_discover_func` as well, because the discovery was not stopped even if GTBS was found and there was no space left for TBS.

MariuszSkamra committed:
Bluetooth: audio: tbs_client: Make GTBS client only build option

Refactor Kconfig to add a way to enable GTBS only client.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

MariuszSkamra committed:
Bluetooth: audio: tbs_client: Fix primary service discovery function

Fixes the discovery function that handles GTBS only discovery. The
discovery stops when there is no space left for another instance. The
function has been split to improve the code readability and avoid
unnecessary UUID comparison.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

Thalley commented on subsys/bluetooth/audio/Kconfig.tbs:
```diff
@@ -165,6 +165,7 @@ config BT_TBS_CLIENT_MAX_CALLS
 config BT_TBS_CLIENT_MAX_TBS_INSTANCES
 	int "Maximum number of TBS instances to setup"
 	default 1
+	range 0 3 if BT_TBS_CLIENT_GTBS
 	range 1 3
```
We can make the range unconditionally 0-3 (or higher) if we decided to always require `BT_TBS_CLIENT_GTBS=y`. 

In other words, apply the same requirement to support GTBS for the client, as the server has. 

That would essentially imply that `BT_TBS_CLIENT_GTBS` is always true, and can be removed.

MariuszSkamra replied:
According to CCP specification, it's up to implementation to use GTBS or TBS.
`Call Control Clients can access TBS, GTBS, or both on a Call Control Server.`
So I guess we cannot imply using GTBS by default

Thalley replied:
But is there a real use case where we want to to have a client implementation that only supports discovering TBS, and not GTBS?

MariuszSkamra replied:
I guess there might be. e.g. Client that can control Skype only

Thalley replied:
Good point :) 

We can leave it as it then
